### ISSUE TITLE
Persistent focus outline after moving a block

### ISSUE THREAD

**Issue by draganescu** [WordPress/gutenberg#28742]
## Description

When moving a block there is an outline that appears around the moved block. The problem is that the outline does not disappear when another block is selected.

## Step-by-step reproduction instructions

0. Have the current (Feb 4th 2021) Gutenberg master running.
1. Add a post and add some blocks.
2. Select one block and move it using the movers in the block toolbar.
3. Select another block.
4. Notice the persistent selection outline on the previously moved block.

## Expected behaviour

The selection outline should always follow the current focus.

## Actual behaviour

The selection outline remained on an unfocused block.

## Screenshots or screen recording (optional)

https://user-images.githubusercontent.com/107534/106916093-7892f780-670f-11eb-8262-52890fc020e8.mp4

## WordPress information
- WordPress version: Trunk
- Gutenberg version: Master
- Are all plugins except Gutenberg deactivated? Yes
- Are you using a default theme (e.g. Twenty Twenty-One)? Yes

## Device information
- Device: Desktop
- Operating system: OSX
- Browser: Firefox latest

**Comment by getsource** [comment]
This was tested in [a triage session today](https://wordpress.slack.com/archives/C02QB2JS7/p1620710335023300), and several folks were able to reproduce it, so removing `needs-testing`.
